Mjz/110 migrate to prql

mzagaja opened the pull request:
Resolve https://github.com/MAPC/datacommon/issues/110

mzagaja committed:
WIP PRQL Migration

Outstanding issues:

1. Refactor to add token parameter with appropriate token to each request
2. Refactor to switch on each kind of database (geospatial DB or tabular)
3. Use of id column in dataset.js does not work with seq_id in _data_browser table.

mzagaja committed:
Set primaryKey to seq_id

For most datasets we are using seq_id as the primary key (as opposed to Carto’s cartodb_id) so we need to tell Ember this is the case in order for it to be able to load data.

mzagaja committed:
Resolve #110 Migrate to PRQL

* Add db_name attribute to the model so we can check it for the appropraite endpoint
* Pick a token based on the db_name
* Append token to all API requests
* Remove shape field from geospatial requests

ericyoungberg commented on app/routes/datasets.js:
```diff
@@ -16,21 +16,36 @@ export default class extends Route {
   model(params) {
     let dataset = this.modelFor('application').findBy('id', params.dataset_id);
     let yearcolumn = dataset.get('yearcolumn');
+    const prqlSchema = dataset.get('schemaname') === 'tabular' ? 'tabular' : 'mapc';
```
Is there a reason why the `schemaname` attribute can't be the `prqlSchema` value outright?

mzagaja replied:
At one point some the `schemaname` values for geospatial datasets were null/empty so this was a workaround for this, but they all seem to have been updated now.

ericyoungberg commented on app/routes/datasets.js:
```diff
@@ -16,21 +16,36 @@ export default class extends Route {
+        break;
+      default:
+        token = config.database.dsToken
+    }
```
Instead of the classic `switch`, a more concise JS way of doing a conditional assignment would be

```js
const token = config.database[`${datasets.get('db_name')}Token`] || config.database.dsToken;
```

ericyoungberg replied:
You don't have to implement this, just throwing the pattern out there.

mzagaja committed:
Remove unnecessary ternary

Remove unnecessary ternary operator because we can just use the ‘schemaname’ value directly from the database.
